bolt-simple-taproot.md

```diff
@@ -0,0 +1,994 @@
+    `NonceAgg` algorithm to derive the combined nonce. 
+    - Local commitment combined nonce: `NonceAgg(verificationNonce, signingNonce)`
+        - Where `verificationNonce` is the local party's and `signingNonce` is
+          the remote party's.
+    - Remote Commitment combined nonce: `NonceAgg(signingNonce, verificationNonce)`
+        - Where `signingNonce` is the local party's and `verificationNonce` is
+          the remote party's.
+  - the `signature` field MUST be a `musig2` partial signature using the
+    aggregated key defined above (serialized according to the Partial Signature
+    Encoding section), and the two public nonces exchanged in prior messages.
+
+  - MUST generate the attached `signature` using the combined
+    Remote commitment combined nonce it generated in the prior step.
+
+The recipient:
+  - MUST validate that the 
+  - MUST validate the signature as a _partial_ signature based on the
+    `PartialSigVerifyInternal` (using the verification nonce of the local
+    party) algorithm of `bip-musig2`:
+
+    - if the partial signature is invalid, MUST fail the channel
+
+#### `funding_signed` Extensions
+
+The sender MUST set:
+
+  - `signature` to the valid signature, using its `funding_pubkey` for the
+    initial commitment transaction, as defined in this documnt.
+    - this signature MUST be a partial signature as defined by `bip-musig2`.
+
+The recipient:
+
+  - MUST validate the signature as a _partial_ signature based on the
+    `PartialSigVerifyInternal` algorithm of `bip-musig2`:
+
+    - if the partial signature is invalid, MUST fail the channel
+
+#### `funding_locked` Extensions
+
+We add a new TLV field to the `funding_locked` message:
+
+1. type: 36 (`funding_locked`)
+2. data:
+    * ...
+    * [`funding_locked_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `funding_locked_tlvs`
+2. types:
+    1. type: 0 (`verification_musig2_pubnonce`)
+    2. data:
+        * [`musig2_pubnonce`:`nonces`]
+    1. type: 2 (`signing_musig2_pubnonce`)
+    2. data:
+        * [`musig2_pubnonce`:`nonces`]
+
+Similar to the `next_per_commitment_point`, by sending the `musig2_pubnonce`
+value in this message, we ensure that the remote party has our public nonce
+which is required to generate a new commitment signature. Similarly, we can't
+generate a new signature until we receive their `verification_musig2_pubnonce`
+and `signing_musig2_pubnonce`.
+
+##### Requirements
+
+The sender MUST:
+
+ - MUST set `verification_musig2_pubnonce` and `signing_musig2_pubnonce` to a
+   _fresh_ set of public nonces as specified by `bip-musig2`
+
+The receiver MUST:
+
+  - MUST combine their local `pubnonce` and the received `pubnonce` via the
+    `NonceAgg` algorithm specified in the `musig2` BIP to received _two_
+    combined nonces. This combined nonce will be used to generate the next
+    commitment signature via the `Sign` algorithm specified in the `musig2`
+    BIP.
+
+
+### Cooperative Closure
+
+#### `shutdown` Extensions
```

Comment:
if shutdown is being kinda re-done, it would be nice if we could fit something in here to allow for adding extra inputs & outputs, ie PSBT funding flow but for channel closes

Reply:
I agree, in the last spec meeting we discussed using the `interactive-tx` protocol (used in dual funding) to collaboratively build two mutual close transactions in parallel (one for each peer, with its own preferences about feerate and other inputs / outputs). This sounds like the right way to do closing in the future, but it does require implementations to already support `interactive-tx`.

So we may take a first intermediate step with a simpler, smaller change to the closing protocol before we eventually move to `interactive-tx`.

Reply:
Latest version moves the co-op close transaction to always signal RBF. This opens the door to the new simple coop close proposal, and future versions that use `interactive-tx` to make diff RBF'd version to add additional inputs/outputs.